group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/SubscribedTopicMetadata.java

```diff
@@ -68,12 +68,12 @@ public int numPartitions(Uuid topicId) {
      * @param topicId       Uuid corresponding to the partition's topic.
      * @param partition     Partition Id within the topic.
      * @return The set of racks corresponding to the replicas of the topics partition.
-     *         If the topic Id does not exist, an empty set is returned
+     *         If the topic Id does not exist or no partition rack information is available, an empty set is returned.
      */
     @Override
     public Set<String> racksForPartition(Uuid topicId, int partition) {
         TopicMetadata topic = this.topicMetadata.get(topicId);
-        return topic == null ? Collections.emptySet() : topic.partitionRacks().get(partition);
+        return topic == null ? Collections.emptySet() : topic.partitionRacks().getOrDefault(partition, Collections.emptySet());
```

Comment:
Should we add a test to SubscribedTopicMetadataTest for this one?

Reply:
good idea, I'll add for numPartitions as well, thanks!